File: logolicious/src/main/java/com/olav/logolicious/screens/fragments/TipFourth.java
package com.olav.logolicious.screens.fragments;

import android.net.Uri;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;

import com.olav.logolicious.R;
import com.olav.logolicious.screens.activities.Activity_Tip;
import com.olav.logolicious.util.GlobalClass;

public class TipFourth extends Fragment{
	
	static int layResId;
	static int fragNum;
	static ImageView imageView1 = null;
	static View v = null;
	@Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        v = inflater.inflate(layResId, container, false);
        ImageButton btnOk = (ImageButton) v.findViewById(R.id.tipOk);
        btnOk.setOnClickListener(new OnClickListener() {
			
			@Override
			public void onClick(View v) {
				if(fragNum == 4)
					getActivity().finish(); // last Tips fragment
				else
					((Activity_Tip)getActivity()).pager.setCurrentItem(fragNum);
			}
		});

        imageView1 = (ImageView) v.findViewById(R.id.imageView1);
        GlobalClass.imageLoader.displayImage(Uri.parse("drawable://" + R.drawable.tip_4_guide).toString(), imageView1);
        return v;
    }
	
    public static TipFourth newInstance(String text, int fragNum, int layResId, int drawableId) {

    	TipFourth.layResId = layResId;
    	TipFourth.fragNum = fragNum;
    	TipFourth f = new TipFourth();
    	
        Bundle b = new Bundle();
        b.putString("msg", text);
        b.putInt("drawable", drawableId);

        f.setArguments(b);
        return f;
    }
    
}

File: logolicious/src/main/java/com/olav/logolicious/screens/fragments/TipFourth.kt
package com.olav.logolicious.screens.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.ImageView
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import com.olav.logolicious.R
import com.olav.logolicious.screens.activities.Activity_Tip
import com.olav.logolicious.util.GlobalClass

class TipFourth : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        v = inflater.inflate(layResId, container, false)
        val btnOk = v?.findViewById<View>(R.id.tipOk) as ImageButton
        btnOk.setOnClickListener {
            if (fragNum == 4) requireActivity().finish() // last Tips fragment
            else (activity as Activity_Tip?)!!.pager!!.currentItem = fragNum
        }
        imageView1 = v?.findViewById<View>(R.id.imageView1) as? ImageView
        imageView1?.let {
            Glide.with(GlobalClass.getAppContext()).load(R.drawable.tip_4_guide).into(
                it
            )
        }
        return v
    }

    companion object {
        var layResId = 0
        var fragNum = 0
        var imageView1: ImageView? = null
        var v: View? = null
        fun newInstance(text: String?, fragNum: Int, layResId: Int, drawableId: Int): TipFourth {
            Companion.layResId = layResId
            Companion.fragNum = fragNum
            val f = TipFourth()
            val b = Bundle()
            b.putString("msg", text)
            b.putInt("drawable", drawableId)
            f.arguments = b
            return f
        }
    }
}